### PR TITLE
BOOKKEEPER-1073: Several stats provider related changes.

### PR DESCRIPTION
**Pull request by sijie**
- add finagle stats provider
- provide the ability to remove gauge and scopes
- update jetty versions for twitter-sciences stats provider

### CODE REVIEW

**Comment by sijie** [comment]
Changes were made from multiple folks at Twitter. I just ported these changes back to the community.

@yzang @leighst

File: bookkeeper-stats-providers/codahale-metrics-provider/src/main/java/org/apache/bookkeeper/stats/CodahaleStatsLogger.java [hunk 1]
```diff
@@ -96,4 +101,9 @@ public StatsLogger scope(String scope) {
         }
         return new CodahaleStatsLogger(metrics, scopeName);
     }
+
+    @Override
+    public void removeScope(String name, StatsLogger statsLogger) {
+        // no-op. the codahale stats logger doesn't the references for stats logger.
```
Comment (jiazhai) [inline bookkeeper-stats-providers/codahale-metrics-provider/src/main/java/org/apache/bookkeeper/stats/CodahaleStatsLogger.java]: typo, leak a verb word here?

File: bookkeeper-stats-providers/codahale-metrics-provider/src/main/java/org/apache/bookkeeper/stats/CodahaleStatsLogger.java [hunk 2]
```diff
@@ -87,6 +87,11 @@ public T getValue() {
     }
 
     @Override
+    public <T extends Number> void unregisterGauge(String statName, Gauge<T> gauge) {
+        // do nothing right now as the Codahale doesn't support conditional removal
```
Comment (jiazhai) [inline bookkeeper-stats-providers/codahale-metrics-provider/src/main/java/org/apache/bookkeeper/stats/CodahaleStatsLogger.java]: Seems metric name is the id for each metric in Codahale, Would a "metrics.remove(metricName)" achieve this unregisterGauge?

Comment (sijie) [inline bookkeeper-stats-providers/codahale-metrics-provider/src/main/java/org/apache/bookkeeper/stats/CodahaleStatsLogger.java]: I think this requires a conditional removal. but codahale doesn't have conditional removal supported.

**Comment by eolivelli** [comment]
@sijie can you rebase

File: bookkeeper-stats-providers/codahale-metrics-provider/src/main/java/org/apache/bookkeeper/stats/CodahaleStatsLogger.java (hunk 1 above)
Comment (sijie) [inline bookkeeper-stats-providers/codahale-metrics-provider/src/main/java/org/apache/bookkeeper/stats/CodahaleStatsLogger.java]: done

**Comment by eolivelli** [comment]
+1 still great.
I am going to merge, or @sijie are you merging?

**Comment by sijie** [comment]
@eolivelli feel free to merge.

**Pull request closed without merging**